event-sourcing/event-subscription-registry/src/test/java/uk/gov/justice/subscription/registry/EventSourceDefinitionRegistryTest.java

```diff
@@ -21,59 +21,134 @@
+
+        final EventSourceDefinition eventSourceDefinition1 = eventSourceDefinition()
+                .withLocation(location)
+                .withName("eventSourceDefinition1")
+                .withDefaultEventSource(true)
+                .build();
 
-        final EventSourceDefinition eventSourceDefinition2 = eventsource()
+        final EventSourceDefinition eventSourceDefinition2 = eventSourceDefinition()
                 .withLocation(mock(Location.class))
-                .withName("eventSourceDefinition1").build();
+                .withName("eventSourceDefinition2")
+                .build();
 
+        final EventSourceDefinitionRegistry eventSourceDefinitionRegistry = new EventSourceDefinitionRegistry();
 
-        final Stream<EventSourceDefinition> eventSourceList = Stream.of(eventSourceDefinition1, eventSourceDefinition2);
+        eventSourceDefinitionRegistry.register(eventSourceDefinition1);
+        eventSourceDefinitionRegistry.register(eventSourceDefinition2);
 
-        final EventSourceDefinitionRegistry eventSourceDefinitionRegistry = new EventSourceDefinitionRegistry(eventSourceList);
 
-        assertThat(eventSourceDefinitionRegistry, is(notNullValue()));
+        final EventSourceDefinition defaultEventSourceDefinition = eventSourceDefinitionRegistry.getDefaultEventSourceDefinition();
 
-        assertThat(eventSourceDefinitionRegistry.getEventSourceDefinitionFor("eventSourceDefinition1").get(), is(sameInstance(eventSourceDefinition1)));
+        assertThat(defaultEventSourceDefinition.getName(), is("eventSourceDefinition1"));
+        assertTrue(defaultEventSourceDefinition.isDefaultEventSource());
     }
 
+    @Test
+    public void shouldThrowExceptionIfSecondDefaultEventSourceIsAdded() {
+
+        final Location location1 = new Location("", "", Optional.of("dataSource"));
+        final Location location2 = new Location("", "", Optional.of("dataSource"));
+
+        final EventSourceDefinition eventSourceDefinition1 = eventSourceDefinition()
+                .withLocation(location1)
+                .withName("eventSourceDefinition1")
+                .withDefaultEventSource(true)
+                .build();
+
+        final EventSourceDefinition eventSourceDefinition2 = eventSourceDefinition()
+                .withLocation(location2)
+                .withDefaultEventSource(true)
+                .withName("eventSourceDefinition2").build();
+
+
+        final EventSourceDefinitionRegistry eventSourceDefinitionRegistry = new EventSourceDefinitionRegistry();
+
+        try {
+            eventSourceDefinitionRegistry.register(eventSourceDefinition1);
+            eventSourceDefinitionRegistry.register(eventSourceDefinition2);
+            fail();
+        } catch (final RegistryException expected) {
+            assertThat(expected.getMessage(), is("You cannot define more than one default event source in event-sources.yaml"));
+        }
+    }
 
```

Comment:
There seems to be a missing test here...previously there was a test for a non existent event source.